### PR TITLE
Mavlink: Update to HEAD from master branch and add upstream updates f…

### PR DESCRIPTION
**Pull request by amilcarlucas**
…rom 20200401

This is to test if the upstream changes break the code
This can only be merged after the #14920

### CODE REVIEW

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -1716,14 +1716,15 @@ void GCS_MAVLINK::send_scaled_imu(uint8_t instance, void (*send_fn)(mavlink_chan
...
+void GCS_MAVLINK::send_scaled_pressure_instance(uint8_t instance, void (*send_fn)(mavlink_channel_t chan, uint32_t time_boot_ms, float press_abs, float press_diff, int16_t temperature, int16_t temperature_press_diff))
 {
     const AP_Baro &barometer = AP::baro();
 
     bool have_data = false;
 
     float press_abs = 0.0f;
-    float temperature = 0.0f;
+    float temperature = 0.0f; // Absolute pressure temperature
+    float temperature_press_diff = 0.0f; // TODO: Differential pressure temperature
```
Comment (IamPete1) [inline libraries/GCS_MAVLink/GCS_Common.cpp:1727]: This should be UINT16_MAX if we don't have it

Comment (WickedShell) [reply libraries/GCS_MAVLink/GCS_Common.cpp:1727]: It's an extension though, so you can't tell if 0 is actually 0C or max... Usually with extensions you have to special case that 0 means unknown. (And if we don't have it a value of 0 is more amenable to packet trimming)

Comment (IamPete1) [reply libraries/GCS_MAVLink/GCS_Common.cpp:1727]: Hum, your right, I was lead astray ;) , **https://github.com/mavlink/mavlink/pull/1429#issuecomment-663314776** Can change the comment to say 0 is the special case I guess.

**Comment by amilcarlucas** [comment]
This is pending some other mavlink and ardupilot PR'S. Do not merge yet

**Comment by amilcarlucas** [comment]
The other PRs have been merged, this one can be merged now

**Comment by amilcarlucas** [comment]
@tridge all test passed, this one is also ready to go in

**Comment by tridge** [comment]
I've merged the dependent submodule change and force pushed this for CI

**Comment by amilcarlucas** [comment]
All CI tests pass now. Thanks Tridge

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**